components/selection/selection-mixin.js

```diff
@@ -152,29 +154,41 @@ export const SelectionMixin = superclass => class extends RtlMixin(CollectionMix
 	}
 
 	_handleRadioKeyUp(e) {
+
 		// check composed path for radio (e.target could be d2l-list-item or other element due to retargeting)
 		if (!e.composedPath()[0].classList.contains('d2l-selection-input-radio')) return;
 		if (e.keyCode < keyCodes.LEFT || e.keyCode > keyCodes.DOWN) return;
 
-		const selectables = Array.from(this._selectionSelectables.values())
-			.filter(item => !item.disabled);
-		let currentIndex = selectables.findIndex(selectable => selectable.selected);
-		if (currentIndex === -1) currentIndex = 0;
-		let newIndex;
-
-		if ((this.dir !== 'rtl' && e.keyCode === keyCodes.RIGHT)
-			|| (this.dir === 'rtl' && e.keyCode === keyCodes.LEFT)
-			|| e.keyCode === keyCodes.DOWN) {
-			if (currentIndex === selectables.length - 1) newIndex = 0;
-			else newIndex = currentIndex + 1;
-		} else if ((this.dir !== 'rtl' && e.keyCode === keyCodes.LEFT)
-			|| (this.dir === 'rtl' && e.keyCode === keyCodes.RIGHT)
-			|| e.keyCode === keyCodes.UP) {
-			if (currentIndex === 0) newIndex = selectables.length - 1;
-			else newIndex = currentIndex - 1;
+		const getSelectionInput = (focusable, forward) => {
```

Comment:
The next focusable must be:
1. A `d2l-selection-input` (radio)
2. Enabled
3. Visible
4. Subscribed with the SelectionMixin (don't move focus to a nested list)

If we don't find one within the selection component's sub-tree, then we need to cycle focus first/last